apis/v1alpha2/validation/httproute_test.go

```diff
@@ -189,134 +175,219 @@ func TestValidateHTTPRoute(t *testing.T) {
+										{
+											Type: gatewayv1a2.HTTPRouteFilterRequestMirror,
+											RequestMirror: &gatewayv1a2.HTTPRequestMirrorFilter{
+												BackendRef: gatewayv1a2.BackendObjectReference{
+													Name: testService,
+													Port: pkgutils.PortNumberPtr(8080),
+												},
 											},
 										},
 									},
 								},
-								{
-									Type: gatewayv1a2.HTTPRouteFilterRequestMirror,
-									RequestMirror: &gatewayv1a2.HTTPRequestMirrorFilter{
-										BackendRef: gatewayv1a2.BackendObjectReference{
-											Name: specialService,
-											Port: portNumberPtr(8080),
-										},
-									},
-								},
 							},
 						},
 					},
 				},
 			},
-			errCount: 2,
+			errCount: 0,
 		},
 		{
-			name: "valid httpRoute with duplicate ExtensionRef filters",
+			name: "invalid httpRoute Rules backendref filters",
 			hRoute: gatewayv1a2.HTTPRoute{
 				Spec: gatewayv1a2.HTTPRouteSpec{
 					Rules: []gatewayv1a2.HTTPRouteRule{
 						{
-							Matches: []gatewayv1a2.HTTPRouteMatch{
-								{
-									Path: &gatewayv1a2.HTTPPathMatch{
-										Type:  pathMatchTypePtr("Prefix"),
-										Value: utilpointer.String("/"),
-									},
-								},
-							},
-							Filters: []gatewayv1a2.HTTPRouteFilter{
+							BackendRefs: []gatewayv1a2.HTTPBackendRef{
 								{
-									Type: gatewayv1a2.HTTPRouteFilterRequestHeaderModifier,
-									RequestHeaderModifier: &gatewayv1a2.HTTPRequestHeaderFilter{
-										Set: []gatewayv1a2.HTTPHeader{
-											{
-												Name:  "special-header",
-												Value: "foo",
+									Filters: []gatewayv1a2.HTTPRouteFilter{
+										{
+											Type: gatewayv1a2.HTTPRouteFilterRequestMirror,
+											RequestMirror: &gatewayv1a2.HTTPRequestMirrorFilter{
+												BackendRef: gatewayv1a2.BackendObjectReference{
+													Name: testService,
+													Port: pkgutils.PortNumberPtr(8080),
+												},
 											},
 										},
-									},
-								},
-								{
-									Type: gatewayv1a2.HTTPRouteFilterRequestMirror,
-									RequestMirror: &gatewayv1a2.HTTPRequestMirrorFilter{
-										BackendRef: gatewayv1a2.BackendObjectReference{
-											Name: testService,
-											Port: portNumberPtr(8080),
+										{
+											Type: gatewayv1a2.HTTPRouteFilterRequestMirror,
+											RequestMirror: &gatewayv1a2.HTTPRequestMirrorFilter{
+												BackendRef: gatewayv1a2.BackendObjectReference{
+													Name: specialService,
+													Port: pkgutils.PortNumberPtr(8080),
+												},
+											},
 										},
 									},
 								},
-								{
-									Type: "ExtensionRef",
-								},
-								{
-									Type: "ExtensionRef",
-								},
-								{
-									Type: "ExtensionRef",
-								},
 							},
 						},
 					},
 				},
 			},
-			errCount: 0,
+			errCount: 1,
 		},
 	}
+
 	for _, tt := range tests {
-		// copy variable to avoid scope problems with ranges
 		tt := tt
```

Comment:
The feedback from the original review is that we should avoid this. Kubernetes golangci config does not have an issue with this config. I can't figure out if they're just using the defaults for golangci, or if I just can't find config attached to this: https://github.com/kubernetes/kubernetes/blob/master/hack/verify-golangci-lint.sh. I think @jpeach is probably the most familiar with our config here - if I'm remembering correctly, our config is intentionally more strict than other k8s projects. 

As a short term workaround for this issue, it looks like we could specifically exclude this lint error: https://github.com/kyoh86/scopelint/issues/4#issuecomment-525540030

Reply:
dcb955a6c679b58cb4e34886940eeab79154916b